ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -1051,7 +1062,8 @@ export default class ConfirmTransactionBase extends Component {
             !valid ||
             submitting ||
             hardwareWalletRequiresConnection ||
-            (gasIsLoading && !gasFeeIsCustom)
+            (gasIsLoading && !gasFeeIsCustom) ||
+            isSigningOrSubmitting
           }
```

Comment:
Should we move this complex disabled logic to a separate place to make it clearer?

Reply:
Maybe in a future refactor, but the goal here is a light touch to facilitate this bug fix, meaning I want to avoid additional risk with changes to legacy code.

Reply:
Hey @matthewwalsh0, quick question: Why did we add `isSigningOrSubmitting`? Doesn't it become redundant since `valid` turns false when `isSigningOrSubmitting` is true?
https://github.com/MetaMask/metamask-extension/blob/950dcc847b117972b55c93f9a7d18327d9718fea/ui/pages/confirm-transaction-base/confirm-transaction-base.component.js#L986